### PR TITLE
Internal: remove react-document-events

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None

**Description**
It is no longer used, has no types, and looks unmaintained.

### CODE REVIEW

**Comment by defunctzombie** [comment]
Blocked by https://github.com/foxglove/studio/pull/1805

File: packages/studio-base/src/components/Panel.tsx [hunk 1]
```diff
@@ -430,12 +430,16 @@ export default function Panel<
...
     );
 
-    const onBlurDocument = useCallback(() => {
-      exitFullScreen();
-      setCmdKeyPressed(false);
-      setShiftKeyPressed(false);
-      onReleaseQuickActionsKey();
-    }, [exitFullScreen, onReleaseQuickActionsKey]);
+    /* Ensure user exits full-screen mode when leaving window, even if key is still pressed down */
+    const visibilityState = useVisibilityState();
```
Comment (jtbandes) [inline packages/studio-base/src/components/Panel.tsx:434]: I think blur is different from visibilityState==="hidden"; is that change intentional?

Comment (defunctzombie) [reply packages/studio-base/src/components/Panel.tsx:434]: yea - you are right - I was trying stuff out. No this should stick with blur if we want this functionality. I'm just trying to rid us of this random module that has this only use.

Comment (defunctzombie) [reply packages/studio-base/src/components/Panel.tsx]: changed to use the tried and true useEffect

**Review by jtbandes — APPROVED** [review]
